addons/survey/models/survey_question.py

```diff
@@ -193,6 +210,20 @@ def _compute_save_as_nickname(self):
             if question.question_type != 'char_box':
                 question.save_as_nickname = False
 
+    @api.depends('is_conditional')
+    def _compute_conditional_question_id(self):
+        for question in self:
+            if not question.is_conditional or question.conditional_question_id is None:
+                question.conditional_question_id = False
+
+    @api.depends('conditional_question_id')
+    def _compute_conditional_answer_id(self):
+        for question in self:
+            if not question.conditional_question_id \
```

Comment:
Well not sure to understand the condition.

Reply:
is it more clear with the docstring ? :)